samcli/lib/build/app_builder.py

```diff
@@ -412,9 +412,13 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
             "platform": get_docker_platform(architecture),
             "rm": True,
         }
+
         if docker_build_target:
             build_args["target"] = cast(str, docker_build_target)
 
+        if self._container_manager and self._container_manager.docker_network_id:
+            build_args["network_mode"] = cast(str, self._container_manager.docker_network_id)
```

Comment:
I think the provided `docker-network` parameter is related to the build container, and not the docker image we create for the lambda function. I believe the problem is the customer is using `host` as docker network, and for some reason that I do not remember, we do not update the created docker network if the provided value is host. check this code snippet:
https://github.com/aws/aws-sam-cli/blob/8bfda3287c617d3fadd0bf670d75bb16e8d07dd6/samcli/local/docker/container.py#L217-L224

Reply:
During offline discussion, it was discussed that building images is likely fine with this change, however building resources under the `-u/--use-container` flag probably has the same issue where the `--network-mode host` does not do anything since the linked logic also binds ports, which is mutually exclusive.